deepmd/pt/model/descriptor/se_r.py

```diff
@@ -0,0 +1,382 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+from typing import (
+    Dict,
+    List,
+    Optional,
+    Tuple,
+)
+
+import numpy as np
+import torch
+
+from deepmd.pt.model.descriptor import (
+    Descriptor,
+    prod_env_mat_se_r,
+)
+from deepmd.pt.utils import (
+    env,
+)
+from deepmd.pt.utils.env import (
+    PRECISION_DICT,
+    RESERVED_PRECISON_DICT,
+)
+from deepmd.pt.utils.env_mat_stat import (
+    EnvMatStatSeR,
+)
+from deepmd.utils.env_mat_stat import (
+    StatItem,
+)
+from deepmd.utils.path import (
+    DPPath,
+)
+
+try:
+    from typing import (
+        Final,
+    )
+except ImportError:
+    pass
+
+from deepmd.dpmodel.utils import EnvMat as DPEnvMat
+from deepmd.pt.model.network.mlp import (
+    EmbeddingNet,
+    NetworkCollection,
+)
+from deepmd.pt.model.network.network import (
+    TypeFilter,
+)
+from deepmd.pt.utils.exclude_mask import (
+    PairExcludeMask,
+)
+
+
+@Descriptor.register("se_e2_r")
+class DescrptSeR(Descriptor):
+    def __init__(
+        self,
+        rcut,
+        rcut_smth,
+        sel,
+        neuron=[25, 50, 100],
+        set_davg_zero: bool = False,
+        activation_function: str = "tanh",
+        precision: str = "float64",
+        resnet_dt: bool = False,
+        exclude_types: List[Tuple[int, int]] = [],
+        old_impl: bool = False,
+        **kwargs,
+    ):
+        super().__init__()
+        self.rcut = rcut
+        self.rcut_smth = rcut_smth
+        self.neuron = neuron
+        self.filter_neuron = self.neuron
+        self.set_davg_zero = set_davg_zero
+        self.activation_function = activation_function
+        self.precision = precision
+        self.prec = PRECISION_DICT[self.precision]
+        self.resnet_dt = resnet_dt
+        self.old_impl = False  # this does not support old implementation.
```

Comment:
I removed the old `implementation` because in the `TypeFilter` class, the output_dim is hard coded as `4`. If we want to keep the old implementation, we need to parameterize the output_dim, or slice the output after the forward net.